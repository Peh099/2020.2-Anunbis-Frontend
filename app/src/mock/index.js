import MockAdapter from 'axios-mock-adapter';
import api from '../services/Api';


const mock = new MockAdapter(api);

export default mock