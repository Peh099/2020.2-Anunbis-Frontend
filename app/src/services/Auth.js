import api from './Api'

export function isAuthenticated(){
  return getToken() !== undefined;
}

export function isProfessor(){
  if(getToken()){
    return localStorage.getItem('professor') ? true : false;
  }
}

export function isStudent(){
  if(getToken()){
    return localStorage.getItem('student') ? true : false;
  }
}

export function getToken() {
  return (localStorage.getItem('access_token'));
}

export async function sendLogin(email, password, callback, errorCallback) {
  const body = {
    email: email,
    password: password
  }

  function writeUser(data){
    localStorage.setItem('access_token', data.access_token);
    if (data.user.reg_student)
      localStorage.setItem('student', JSON.stringify(data.user));
    else 
      localStorage.setItem('professor', JSON.stringify(data.user));
  }

  api.post("/login", body)
    .then(response => {
      const data = response.data;
      if (data.access_token)
        writeUser(data);
      else
        logOut();
    })
    .then(() => callback())
    .catch(() => errorCallback())
}

export function logOut() {
  localStorage.removeItem('access_token');
  localStorage.removeItem('student');
  localStorage.removeItem('professor');
}