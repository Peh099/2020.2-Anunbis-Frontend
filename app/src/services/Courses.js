import api from './Api';

export const getCourses = (setCourses) => {
  api.get('/course').then((response) => {
    console.log(response);
    setCourses(response.data);
  });
};
