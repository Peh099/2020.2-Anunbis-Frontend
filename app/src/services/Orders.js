export function orderPop(post1, post2) {
  if (post1?.feedbacks.agrees > post2?.feedbacks.agrees) return -1;
  if (post1?.feedbacks.agrees < post2?.feedbacks.agrees) return 1;
  return 0;
}

export function orderDate(post1, post2) {
  if (post1?.post_date > post2?.post_date) return -1;
  if (post1?.post_date < post2?.post_date) return 1;
  return 0;
}

export function orderHRate(post1, post2) {
  if (post1?.rating > post2?.rating) return -1;
  if (post1?.rating < post2?.rating) return 1;
  return 0;
}

export function orderLRate(post1, post2) {
  if (post1?.rating < post2?.rating) return -1;
  if (post1?.rating > post2?.rating) return 1;
  return 0;
}

export const orders = [
  { fun: orderPop, id: 0, name: 'Mais popular', selected: true },
  { fun: orderDate, id: 1, name: 'Mais recente' },
  { fun: orderHRate, id: 2, name: 'Maior Nota Geral' },
  { fun: orderLRate, id: 3, name: 'Menor Nota Geral' },
];
