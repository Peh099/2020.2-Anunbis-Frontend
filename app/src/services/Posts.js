import api from './Api';

export const getPosts = (setPosts) => {
  api.get('post').then((response) => {
    if (response.status === 200) setPosts(response.data);
  });
};
