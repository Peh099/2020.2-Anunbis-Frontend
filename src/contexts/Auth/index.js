import { React, createContext, useState } from "react";

const AuthContext = createContext({});

export function AuthProvider({children}) {
    const [user, setUser] = useState([]); //<object | null>(null);
    const [token, setToken] = useState([]);

    function Login(props){
        setUser(props.user);
        setToken(props.token);
    }
    return(
        <AuthContext.Provider value={{ signed: Boolean(user), user, token, Login }}>
            {children}
        </AuthContext.Provider>
    )
}

export default AuthContext;