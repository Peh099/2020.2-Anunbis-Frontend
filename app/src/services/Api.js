import axios from 'axios';
import { getToken } from './Auth';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_HOST,
});

api.interceptors.request.use(
  (config) => {
    const accessToken = getToken();
    config.headers.Authorization = `Bearer ${accessToken}`;
    return config;
  },
  (error) => {
    console.log(error);
  },
);

export default api;
