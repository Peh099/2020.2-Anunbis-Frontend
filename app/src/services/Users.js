import { getToken } from './Auth';

const createUser = (localStorageName, homePath) => {
    return {
        localStorageName: localStorageName,
        homePath: homePath,
        isAuthenticated: () => getToken() ? localStorage.getItem(localStorageName) ? true : false : false
    }
}

const Users = {
    PROFESSOR: createUser("professor", '/professor/'),
    STUDENT: createUser('student', '/student/'),
    VISITANT: createUser('visitant', '/'),
    isAuthenticated: () => getToken() !== null,
    whoAuthenticated: function () {
        if (getToken()) {
            const userName = Object.keys(this).find((user) => 
                localStorage.getItem(this[user]?.localStorageName) !== null
            )
            return this[userName];
        }
        return this.VISITANT;
    }
}

export default Users;