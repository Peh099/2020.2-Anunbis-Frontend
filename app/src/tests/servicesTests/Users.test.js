import renderer from 'react-test-renderer';
import Users, { getToken } from '../../services/Users';
import { sendLogin } from '../../services/Auth';
import mock from '../../mock';
import { waitFor } from '@testing-library/react';
import { validStudent, validProfessor } from '../../mock/fixtures/stored_users';
import mockLogin from '../../mock/fixtures/login';

describe('Test Api', () => {
  it('must return professor for the all functions', async () => {
    const callback = jest.fn();

    mockLogin('student', 200);

    sendLogin(validStudent.email, validStudent.password, callback, {});

    await waitFor(() => {
      expect(callback).toHaveBeenCalled();
    });

    expect(Users.isAuthenticated()).toBe(true);
    expect(Users.STUDENT.isAuthenticated()).toBe(true);
    expect(Users.STUDENT.data()).toEqual(validStudent.response);
    expect(Users.whoAuthenticated()).toEqual(Users.STUDENT);
  });
  it('must return professor for the all functions', async () => {
    const callback = jest.fn();

    mock.reset();
    mockLogin('professor', 200);

    sendLogin(validProfessor.email, validProfessor.password, callback, {});

    await waitFor(() => {
      expect(callback).toHaveBeenCalled();
    });

    expect(Users.isAuthenticated()).toBe(true);
    expect(Users.PROFESSOR.isAuthenticated()).toBe(true);
    expect(Users.PROFESSOR.data()).toEqual(validProfessor.response);
    expect(Users.whoAuthenticated()).toEqual(Users.PROFESSOR);
  });
});
