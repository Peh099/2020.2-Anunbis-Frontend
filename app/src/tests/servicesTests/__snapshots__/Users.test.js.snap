// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Api component matches the snapshot 1`] = `null`;

exports[`Snapshot Api component matches the snapshot 2`] = `
Object {
  "PROFESSOR": Object {
    "data": [Function],
    "homePath": "/professor/",
    "isAuthenticated": [Function],
    "localStorageName": "professor",
  },
  "STUDENT": Object {
    "data": [Function],
    "homePath": "/student/",
    "isAuthenticated": [Function],
    "localStorageName": "student",
  },
  "VISITANT": Object {
    "data": [Function],
    "homePath": "/",
    "isAuthenticated": [Function],
    "localStorageName": "visitant",
  },
  "isAuthenticated": [Function],
  "whoAuthenticated": [Function],
}
`;
